share/shwap/p2p/shrex/shrex_getter/shrex.go

```diff
@@ -187,13 +191,21 @@ func (sg *Getter) GetEDS(ctx context.Context, header *header.ExtendedHeader) (*r
 
 		reqStart := time.Now()
 		reqCtx, cancel := utils.CtxWithSplitTimeout(ctx, sg.minAttemptsCount-attempt+1, sg.minRequestTimeout)
-		eds, getErr := sg.edsClient.RequestEDS(reqCtx, header.DAH, header.Height(), peer)
+		getErr = sg.ndClient.Get(reqCtx, &request, response, peer)
 		cancel()
 		switch {
```

Comment:
`&request` why we need pointer here? Would it work jsut fine with value because all `id` have value receiver methods?

Reply:
Since IDs have both pointer and value receivers. Receviers will be fixed subsequently.

Reply:
`NewEDSID{}` creates a raw value and not a pointer.